### ISSUE TITLE
signal SIGSEGV: segmentation violation

### ISSUE THREAD

**Issue by adoniscyp** [dapr/dapr#6223]
## In what area(s)?

/area runtime

## What version of Dapr?
Runtime version: 1.10.0

## Behavior

We use Dapr for a few of our microservices and recently one of the services pods keeps restarting. We notice the following in Dapr logs. Pods of this service will keep restarting every 3 hours or so

```
 panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x3966d83]

goroutine 364 [running]:
github.com/dapr/dapr/pkg/messaging/v1.(*InvokeMethodResponse).Headers(...)
	/home/runner/work/dapr/dapr/pkg/messaging/v1/invoke_method_response.go:112
github.com/dapr/dapr/pkg/http.(*api).onDirectActorMessage(0xc000af2600, 0xc0010a1200)
	/home/runner/work/dapr/dapr/pkg/http/api.go:1754 +0x9c3
github.com/dapr/dapr/pkg/http.(*server).unescapeRequestParametersHandler.func1(0xc0010a1200)
	/home/runner/work/dapr/dapr/pkg/http/server.go:319 +0x71
github.com/fasthttp/router.(*Router).Handler(0xc000aa0a10, 0xc0010a1200)
	/home/runner/go/pkg/mod/github.com/fasthttp/router@v1.4.12/router.go:414 +0x46a
github.com/dapr/dapr/pkg/diagnostics.(*httpMetrics).FastHTTPMiddleware.func1(0xc0010a1200)
	/home/runner/work/dapr/dapr/pkg/diagnostics/http_monitoring.go:227 +0x13b
github.com/dapr/dapr/pkg/diagnostics.HTTPTraceMiddleware.func1(0xc0010a1200)
	/home/runner/work/dapr/dapr/pkg/diagnostics/http_tracing.go:56 +0x1ad
github.com/dapr/dapr/pkg/http.(*server).apiLoggingInfo.func1(0xc0010a1200)
	/home/runner/work/dapr/dapr/pkg/http/server.go:238 +0x22c
github.com/valyala/fasthttp.(*Server).serveConn(0xc000de8d80, {0x5161ec0?, 0xc00057f408})
	/home/runner/go/pkg/mod/github.com/valyala/fasthttp@v1.40.0/server.go:2311 +0x1268
github.com/valyala/fasthttp.(*workerPool).workerFunc(0xc000e003c0, 0xc001349ae0)
	/home/runner/go/pkg/mod/github.com/valyala/fasthttp@v1.40.0/workerpool.go:224 +0xa9
github.com/valyala/fasthttp.(*workerPool).getCh.func1()
	/home/runner/go/pkg/mod/github.com/valyala/fasthttp@v1.40.0/workerpool.go:196 +0x38
created by github.com/valyala/fasthttp.(*workerPool).getCh
	/home/runner/go/pkg/mod/github.com/valyala/fasthttp@v1.40.0/workerpool.go:195 +0x1b0
```

We are on EKS 1.24. We upgraded EKS from 1.23 to 1.24 a couple of weeks ago but i think EKS version is not relevant to this issue since it only happens to one of our services.

**Comment by adoniscyp** [comment]
Thanks for the quick response @ItalyPaleAle 

I did a cursory search in our code base and i don't think we are using `ServiceInvocationStreaming`. I will upgrade to 1.10.4 and report back

**Comment by adoniscyp** [comment]
Issue seems fixed after the upgrade to 1.10.4 and monitoring for 6 hours. I will reopen if I notice it is happening again

**Comment by ItalyPaleAle** [comment]
Thanks for confirming.

However, I am still a bit puzzled as to why that was happening. Looking at the stack trace, it doesn't match what I see in the code for the lines that are indicated, at the 1.10.0 tag.

Can you confirm you were using 1.10.0 before?

**Comment by adoniscyp** [comment]
I think i might have been on another version when i opened this ticket and not on 1.10.4. I reported the output of` dapr version`

The thing is there seem to be another bug with the CLI. Currently I am on 1.10.4 and: 
`dapr version 
CLI version: 1.9.1 
Runtime version: 1.10.0`

When i upgraded to 1.10.4 earlier in the day i was puzzled why the cli was reporting 1.10.0 but the worked without errors. I checked the pod metadata and the upgrade was succesfull to 1.10.4.
